### PR TITLE
GMX GLP Spells

### PR DESCRIPTION
**Pull request by 1chioku**
Brief comments on the purpose of your changes:
1. Updated arbitrum_base_sources.yml to support erc20_arbitrum.evt_transfer
2. Created models/gmx/arbitrum/glp , and I added 4 spells in this folder.
3. Created models/gmx/arbitrum/vault , and I added 1 spells in this folder.

*For Dune Engine V2*
I've checked that:
General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributors)
* [x] if adding a new materialized table, I edited the optimize table macro

Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

File: models/gmx/arbitrum/glp/gmx_arbitrum_glp_components.sql [hunk 1]
```diff
@@ -0,0 +1,926 @@
+{{ config(
+        alias = 'components',
+        materialized = 'incremental',
+        file_format = 'delta',
+        incremental_strategy = 'merge',
+        unique_key = ['minute'],
```
Comment (soispoke) [inline models/gmx/arbitrum/glp/gmx_arbitrum_glp_components.sql:6]: Are you 100% sure there are never going to be any duplicates if we only use `'minute'` as the unique key ? Would it make sense to add the `'amount'` in there as well ?

**Review by soispoke — COMMENTED** [review]
Thanks for your submission @1chioku ! It's very impressive work and there only very small details/comments I had.

The last thing needed is to add gmx/arbitrum to the dbt_project.yml file, and we're good to go !

File: models/gmx/arbitrum/glp/gmx_arbitrum_glp_sources.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+version: 2
+
+sources: 
+  - name: gmx_arbitrum
+    freshness:
+          warn_after: { count: 12, period: hour }
+          error_after: { count: 24, period: hour }
```
[the remark this answers is not in the document]
Comment (1chioku) [reply models/gmx/arbitrum/glp/gmx_arbitrum_glp_sources.yml:7]: done

File: models/gmx/arbitrum/glp/gmx_arbitrum_glp_components.sql (hunk 1 above)
Comment (1chioku) [reply models/gmx/arbitrum/glp/gmx_arbitrum_glp_components.sql:6]: The query uses: [`explode(sequence(TIMESTAMP '2021-08-31 08:13', CURRENT_TIMESTAMP, INTERVAL 1 minute))`](https://github.com/duneanalytics/spellbook/pull/1665/files#diff-35ece43ccefc27bae1364e223e1bb0fccf6c3920f62c3b47e5c8e17319078bb9R16) to generate the primary key for this table, and all subsequent data manipulations are `LEFT JOINED` into the the minute column, so there should not be any duplicates.

Apart from the minute column, there is no column with unique data in this abstraction.

Furthermore I added a unit test [here](https://github.com/duneanalytics/spellbook/pull/1665/files#diff-9dad9abe64cc3cebb8cdefdfd4094ef06f9042b2fdd56a6e1560960768889b01R18) that test the minute column for uniqueness, does that help?

So I cant add a second unique key as such, but. I can concatenate 2 random column like this : `unique_key="minute||'-'||weth_available_assets"` to generate a stronger unique_key?

Let me know if you want me to do same.

Comment (soispoke) [reply models/gmx/arbitrum/glp/gmx_arbitrum_glp_components.sql:6]: @1chioku No it's ok, I tested it a couple of times and it looks good to me
Just add your info in the dbt_project.yml file please and we're good 👍

**Review by soispoke — APPROVED** [review]

**Pull request merged**